halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/canary/prometheus/PrometheusCanaryAccountValidator.java

```diff
@@ -55,7 +54,7 @@ public void validate(ConfigProblemSetBuilder p, AbstractCanaryAccount n) {
     if (StringUtils.isNotEmpty(usernamePasswordFile)) {
       String usernamePassword = validatingFileDecrypt(p, usernamePasswordFile);
 
-      if (Strings.isNullOrEmpty(usernamePassword)) {
+      if (usernamePassword != null && StringUtils.isEmpty(usernamePassword)) {
```

Comment:
Shouldn't this `&&` be an `||` to keep the current behavior (or was the change deliberate)?

Reply:
It is hard to guess intent, but here goes...
I'd guess that it should not  be && because that leaves the 'else if' condition the chance of throwing an NPE when usernamePassword is null. Making it || is redundant since StringUtils.isEmpty checks for null and "". Given the context, the only thing that makes sense is for it to just be StringUtils.isEmpty(usernamePassword) unless the surrounding code is further modified. I'm going to make that change.

Reply:
fixed